### PR TITLE
feat: create robot accounts with push creds for teams

### PR DESCRIPTION
**Pull request by srodenhuis**
Creating a Harbor robot account for each team with push credentials for team member to download

### CODE REVIEW

**Review by dennisvankekem — APPROVED** [review]
LGTM

File: src/tasks/harbor/harbor.ts [hunk 1]
```diff
@@ -162,18 +163,70 @@ async function createTeamRobotAccount(projectName: string): Promise<RobotCreated
...
 
-  const robotAccount = (await doApiCall(errors, `Creating robot account ${fullName} with project level perms`, () =>
-    robotApi.createRobot(projectRobot),
+  const robotPullAccount = (await doApiCall(
+    errors,
+    `Creating pull robot account ${fullName} with project level perms`,
+    () => robotApi.createRobot(projectRobot),
+  )) as RobotCreated
+  if (!robotPullAccount?.id) {
+    throw new Error(
```
Comment (j-zimnowoda) [inline src/tasks/harbor/harbor.ts:175]: I am not sure if that is right wait of dealing with this error.
If I cannot create a robot account does it mean I cannot configure other parts of of Harbor ?

File: src/tasks/harbor/harbor.ts [hunk 2]
```diff
@@ -162,18 +163,70 @@ async function createTeamRobotAccount(projectName: string): Promise<RobotCreated
...
+      `RobotPullAccount already exists and should have been deleted beforehand. This happens when more than 100 robot accounts exist.`,
```
Comment (j-zimnowoda) [inline src/tasks/harbor/harbor.ts:176]: Is this a reported bug in harbor ?

File: src/tasks/harbor/harbor.ts [hunk 3]
```diff
@@ -162,18 +163,70 @@ async function createTeamRobotAccount(projectName: string): Promise<RobotCreated
...
+    )
+  }
+  return robotPullAccount
+}
+
+/**
+ * Create Harbor system robot account that is scoped to a given Harbor project
+ * @param projectName Harbor project name
+ */
+async function createTeamPushRobotAccount(projectName: string): Promise<any> {
```
Comment (j-zimnowoda) [inline src/tasks/harbor/harbor.ts:186]: Since it does not always create a secret can we rename this function to `ensureTeamPushRobotAccount` ?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
There is too much duplicated code.

**Review by j-zimnowoda — APPROVED** [review]

**Pull request merged**